server/sonar-server/src/test/java/org/sonar/server/component/ws/SuggestionsActionTest.java

```diff
@@ -109,6 +110,110 @@ public void define_suggestions_action() {
+      .containsExactly(Qualifiers.PROJECT);
+
+    // assert correct id to be found
+    assertThat(response.getResultsList())
+      .flatExtracting(Category::getItemsList)
+      .extracting(Suggestion::getKey, Suggestion::getOrganization)
+      .containsExactly(tuple(project.getKey(), organization.getKey()));
+  }
+
+  @Test
+  public void suggestions_without_query_should_contain_favorites() throws Exception {
+    ComponentDto project = db.components().insertComponent(newPrivateProjectDto(organization));
+    doReturn(singletonList(project)).when(favoriteFinder).list();
+
+    componentIndexer.indexOnStartup(null);
+    authorizationIndexerTester.allowOnlyAnyone(project);
+
+    SuggestionsWsResponse response = actionTester.newRequest()
+      .setMethod("POST")
+      .executeProtobuf(SuggestionsWsResponse.class);
+
+    // assert match in qualifier "TRK"
+    assertThat(response.getResultsList())
+      .filteredOn(q -> q.getItemsCount() > 0)
+      .extracting(Category::getQ)
+      .containsExactly(Qualifiers.PROJECT);
+
+    // assert correct id to be found
+    assertThat(response.getResultsList())
+      .flatExtracting(Category::getItemsList)
+      .extracting(Suggestion::getKey, Suggestion::getOrganization)
+      .containsExactly(tuple(project.getKey(), organization.getKey()));
+  }
+
+  @Test
+  public void suggestions_without_query_should_not_contain_matches_that_are_neither_favorites_nor_recently_browsed() throws Exception {
+    ComponentDto project = db.components().insertComponent(newPrivateProjectDto(organization));
+
+    componentIndexer.indexOnStartup(null);
+    authorizationIndexerTester.allowOnlyAnyone(project);
+
+    SuggestionsWsResponse response = actionTester.newRequest()
+      .setMethod("POST")
+      .executeProtobuf(SuggestionsWsResponse.class);
+
+    // assert match in qualifier "TRK"
+    assertThat(response.getResultsList())
+      .filteredOn(q -> q.getItemsCount() > 0)
+      .extracting(Category::getQ)
+      .isEmpty();
+  }
+
+  @Test
+  public void suggestions_without_query_should_order_results() throws Exception {
+    ComponentDto project1 = db.components().insertComponent(newPrivateProjectDto(organization).setName("Alpha"));
+    ComponentDto project2 = db.components().insertComponent(newPrivateProjectDto(organization).setName("Bravo"));
+    ComponentDto project3 = db.components().insertComponent(newPrivateProjectDto(organization).setName("Charlie"));
+    ComponentDto project4 = db.components().insertComponent(newPrivateProjectDto(organization).setName("Delta"));
+    doReturn(asList(project4, project2)).when(favoriteFinder).list();
+
+    componentIndexer.indexOnStartup(null);
+    authorizationIndexerTester.allowOnlyAnyone(project1);
+    authorizationIndexerTester.allowOnlyAnyone(project2);
+    authorizationIndexerTester.allowOnlyAnyone(project3);
+    authorizationIndexerTester.allowOnlyAnyone(project4);
+
+    SuggestionsWsResponse response = actionTester.newRequest()
```

Comment:
remark: in many places, WsActionTester is named ws in the test. For consistency, I suggest to use the same name